crates/handler/src/validation.rs

```diff
@@ -307,30 +307,62 @@ mod tests {
     use crate::{ExecuteCommitEvm, MainBuilder, MainContext};
     use bytecode::opcode;
     use context::{
-        result::{EVMError, ExecutionResult, HaltReason, InvalidTransaction, Output},
-        Context, TxEnv,
+        result::{EVMError, ExecutionResult, HaltReason, InvalidTransaction, Output}, CfgEnv, Context, TxEnv
     };
     use database::{CacheDB, EmptyDB};
-    use primitives::{address, Address, Bytes, TxKind, MAX_INITCODE_SIZE};
+    use primitives::{address, Address, Bytes, TxKind, eip3860, eip7825, eip7907, hardfork::SpecId};
 
     fn deploy_contract(
         bytecode: Bytes,
+        spec_id: Option<SpecId>,
+        gas_limit: Option<u64>,
     ) -> Result<ExecutionResult, EVMError<core::convert::Infallible>> {
-        let ctx = Context::mainnet().with_db(CacheDB::<EmptyDB>::default());
+        let mut cfg = CfgEnv::default();
+        if let Some(spec_id) = spec_id {
+            cfg.spec = spec_id;
+        }
+        let block_and_txn_gas_limit_cap = gas_limit.unwrap_or(eip7825::TX_GAS_LIMIT_CAP);
+        cfg.tx_gas_limit_cap = Some(block_and_txn_gas_limit_cap);
+
+        let ctx = Context::mainnet().with_db(CacheDB::<EmptyDB>::default()).with_cfg(cfg);
```

Comment:
@rakita I was thinking to add more tests to check the journal/db if code is warmed properly and txns with multiple `CALL` and `CREATE`. Should this be done in this repo and if so are there any good examples of this? or should these scenarios be covered in the `EEST` tests?